pages/desktop/user.py

```diff
@@ -119,8 +124,28 @@ def click_update_account(self):
         self.selenium.find_element(*self._update_account_locator).click()
         WebDriverWait(self.selenium, self.timeout).until(lambda s: self.update_message == "Profile Updated")
 
-    def change_hide_email_state(self):
-        self.selenium.find_element(*self._hide_email_checkbox).click()
+    def hide_email_address(self):
+        email_checkbox = self.selenium.find_element(*self._email_checkbox)
+        WebDriverWait(self.selenium, self.timeout).until(expected_conditions.visibility_of(email_checkbox));
+
+        if email_checkbox.is_selected():
+            err_msg = 'Email is already hidden'
+            return err_msg
+
+        if not email_checkbox.is_selected():
+            self.selenium.find_element(*self._email_checkbox).click()
+        return email_checkbox.is_selected() == True
```

Comment:
Again you are returning something from a method that isn't expected to return anything, so this code won't do anything. I think what we need to do here is instead to wait for the email checkbox to be checked. You can do this as above with `WebDriverWait`, e.g., `WebDriverWait(self.selenium, self.timeout).until(lambda s: email_checkbox.is_selected())`